### ISSUE TITLE
Import animation channels

### ISSUE THREAD

**Issue by bezo97** [bezo97/IFSRenderer#37]
This would enable the integration of IFSRenderer into existing animation pipelines. The goal would be importing a CSV file containing various animation channels, primarily for camera parameters.

The request comes from @vitos1k in [this](https://github.com/bezo97/IFSRenderer/discussions/34) forum discussion.

**Comment by vitos1k** [comment]
Do you need some help with it? Or maybe can guide me where in the code to look for?

**Comment by bezo97** [comment]
@vitos1k Don't worry I'm on it, been busy the last few days. Hoping to have a working version by end of week.

**Comment by bezo97** [comment]
I have solution that can be tested on the `dev` branch. I added an `Import` button to the bottom of the channel list: 
![image](https://github.com/bezo97/IFSRenderer/assets/26039350/e204f41a-d742-41be-907b-1fd177763ea0)

The way I implemented it is that the column names are the channel paths, so any channel can be imported, not just the camera object. @vitos1k this means you'll need to adjust your script a bit, or modify the csv before importing it. Here's a test file [example3.csv](https://github.com/user-attachments/files/15521852/example3.csv) that shows the correct way (with values based off of your example). 

Let me know if this works for you.

**Comment by vitos1k** [comment]
It works, as in it imports data and animation is playing. BUT!  It seems that some rotations are inverted??. 
1. Camera is looking the oposite (180degree) way from exported camera. So i rotated TilitY offset 180 degrees and now it matches. 2. Then it looked like Cameras are pointing the same direction at the begining of the animation, but now it rotates Clockwise in IFSRenderer, while it rotates CounterClockwise in my animation. I made a video comparison from the output from IFSRenderer and Blender(keep  in mind that IFSRenderers camera is rotated 180 degrees via TiltY parameter)  See video
https://github.com/bezo97/IFSRenderer/assets/1842518/f45eac1d-b13f-4ad9-bb7f-95e99a214e46

3. I thought that all the angles would be inverted, but from the looks of it only rotation around vertical axis is inverted. Because i've flipped video from IFSRenderer in horizontal axis and they matched spot on with blenders render


https://github.com/bezo97/IFSRenderer/assets/1842518/7e1f9d89-853e-49b7-9405-1f3bba8ccb64

**Comment by bezo97** [comment]
The problem with the camera orientation is probably caused by the different coordinates systems used in Blender and IFSRenderer. [This is what Blender uses](https://docs.blender.org/manual/en/2.91/editors/3dview/navigate/viewpoint.html#:~:text=X%20axis,Top%20/%20Bottom). And this is how it works in IFSRenderer: 
```
X axis: Left / Right
Y axis: Top / Bottom
Z axis: Front / Back
```
In other words, Blender uses right-handed coordinate system, and IFSRenderer uses left-handed coordinate system. This is something your conversion script could handle, not the import feature of IFSR. [Found a thread](https://stackoverflow.com/questions/1263072/changing-a-matrix-from-right-handed-to-left-handed-coordinate-system) that should help you add this conversion to the script. Ideally you won't need to add any tilting to work correctly. 
Also, I know there's an option in Blender to select the correct axis when exporting objects, maybe you can find a way to reuse the functionality: 
![image](https://github.com/bezo97/IFSRenderer/assets/26039350/1d161c45-c3f3-48cd-b17a-31ceec8c9b02)

As for the channel paths, you are right they aren't shown anywhere. But instead of tooltips I'll try to figure out a way to be able to copy-paste them. In the meantime you can also peek at the saved fractal file to see them.

**Comment by bezo97** [comment]
I made the channel paths visible in these two places: the context menu of the sliders, and the channel editor panel. They are read-only textboxes. 
![image](https://github.com/bezo97/IFSRenderer/assets/26039350/4275e282-85ed-4083-878e-a8997df09e67)

**Comment by vitos1k** [comment]
Thanks alot! I'll try it tomorrow when will be near my machine. And i'll modify my script to take into account left-handedness. Will report back here

**Comment by vitos1k** [comment]
Yep works as expected! i've updated my scrip for blender here https://github.com/vitos1k/pipelinetools/blob/master/io_export_anim.py

Also as a request is it possible to export animation in csv file? For example Create interesting pattern in IFSRenderer, create few keypoints for camera animation, export it to blender, polish animation, add aditional graphics. And then reimport new polished animation into IFSRenderer for final render.

**Comment by bezo97** [comment]
@vitos1k Awesome, thanks for testing it out. The feature will be available in the next release 👍 

As for the CSV export, I feel its a bit out-of-scope but I'll think about it. Feel free to open another Issue or send a PR for this

**Comment by bezo97** [comment]
Closing this as the feature is released in 1.4.1
